pkg/services/ngalert/notifier/channels/googlechat.go

```diff
@@ -98,21 +99,25 @@ func (gcn *GoogleChatNotifier) Notify(ctx context.Context, as ...*types.Alert) (
 	}
 
 	ruleURL := joinUrlPath(gcn.tmpl.ExternalURL.String(), "/alerting/list", gcn.log)
-	// Add a button widget (link to Grafana).
-	widgets = append(widgets, buttonWidget{
-		Buttons: []button{
-			{
-				TextButton: textButton{
-					Text: "OPEN IN GRAFANA",
-					OnClick: onClick{
-						OpenLink: openLink{
-							URL: ruleURL,
+	if gcn.isUrlAbsolute(ruleURL) {
```

Comment:
Could we add a test case for this to the `googlechat_test.go` file next to this one?

All the existing cases provide a valid base URL in the settings (`http://localhost`). So, they never run into this problem.

https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/notifier/channels/googlechat_test.go#L37-L91

It'd be nice to have another case that does not have this in settings, and then asserts that the `expMsg` does not have the `buttonWidget`.

Reply:
I added 2 Testcases to the ngvariant of the channel: one for an empty URL, one for a relative URL. I did not add a testcase for the old alerting